src/wrappers/themis/rust/libthemis-sys/bindgen.sh

```diff
@@ -29,8 +29,12 @@ WHITELIST="(THEMIS|themis|secure_(comparator|session)|STATE)_.*"
 # thing to do is to split the generated code into different files for different
 # platforms (like themis_x86_64.rs, themis_arm64.rs, etc.) and conditionally
 # compile them depending on target.
+#
+# Though, we do disable layout tests which *are* architecture-specific and
+# we cannot include them without breaking either 32-bit or 64-bit machines.
```

Comment:
Perhaps we could include tests depending on the current architecture?

Reply:
> Perhaps we could include tests depending on the current architecture?

We could but there are a lot of _ifs_ here. Since Docker does not support 32-bit hosts, cloud-based CI services do not offer 32-bit runners. We probably could plug in our own into GitHub Actions, but I don't know whether their runner supports 32-bit systems. Ideally we'd be testing on ARM systems as well.

We can use Docker containers with 32-bit userland but that's not quite the same. Though, it'd probably catch the issues we have encountered here since they are more about integer overflow than memory limits.

Another thing to consider is that it would double the test matrix since we'd better be testing _all_ wrappers, Core alone is not enough.

Overall, I'd like to get this done eventually. Right now we rely only on internal BuildBot runs to have something resembling regular testing on something other that x86_64.

---

Or do you mean that layout tests should be there, just be different for architectures?

```
if cfg!(target_pointer_size = "32") {
    // test for 32-bit
} else if cfg!(target_pointer_size = "64") {
    // test for 64-bit
} else {
    // dunno fail
}
```

Well, I'd like Bindgen to be able to generate tests like that automatically, but that's quite hard. The best we could get right now is generate multiple modules:

- `src/x86_64-unknown-linux-gnu.rs`
- `src/i686-unknown-linux-gnu.rs`
- `src/aarch64-apple-ios.rs`
- ...

And then conditionally build and reexport one of them.

However, since _the only_ thing that is different is the tests, I feel that it's a little bit overkill.

Reply:
> Or do you mean that layout tests should be there, just be different for architectures?
Of course, it would be great to have a complete set of test for each of platforms.

Well, I share your view at all. Lets start from this position and have all our future plans in the backlog :)

Reply:
> Lets [...] have all our future plans in the backlog :)

Yup. I've created an internal task to track this (T1655).